src/main/java/github/scarsz/discordsrv/objects/managers/link/file/AppendOnlyFileAccountLinkManager.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * DiscordSRV - https://github.com/DiscordSRV/DiscordSRV
+ *
+ * Copyright (C) 2016 - 2022 Austin "Scarsz" Shapiro
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/gpl-3.0.html>.
+ */
+
+package github.scarsz.discordsrv.objects.managers.link.file;
+
+import github.scarsz.discordsrv.DiscordSRV;
+import github.scarsz.discordsrv.util.DiscordUtil;
+import lombok.SneakyThrows;
+import net.dv8tion.jda.api.entities.User;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.bukkit.Bukkit;
+import org.bukkit.OfflinePlayer;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Map;
+import java.util.UUID;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class AppendOnlyFileAccountLinkManager extends AbstractFileAccountLinkManager {
+
+    // matches "discordId uuid" with anything after https://regex101.com/r/oRiDUP
+    private static final Pattern LINK_PATTERN = Pattern.compile("^(?<discord>\\d+) (?<uuid>[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}).*");
+    // matches "-discordId" or "-uuid" or "-discord uuid" or "-uuid discord" https://regex101.com/r/SqriBn
+    private static final Pattern MODIFICATION_PATTERN = Pattern.compile("^-(?>(?>(?<discord>\\d+)|(?<uuid>[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})) ?)+$");
+
+    public AppendOnlyFileAccountLinkManager() {
+        super();
+    }
+
+    @Override
+    void load() throws IOException {
+        File linkedAccountsJsonFile = new File(DiscordSRV.getPlugin().getDataFolder(), "linkedaccounts.json");
+        if (linkedAccountsJsonFile.exists()) {
+            @SuppressWarnings("deprecation") JsonFileAccountLinkManager manager = new JsonFileAccountLinkManager();
+            manager.getLinkedAccounts().forEach(this::link);
+            int count = manager.getLinkedAccountCount();
+
+            File newFile = new File(DiscordSRV.getPlugin().getDataFolder(), "linkedaccounts.json.delete");
+            if (linkedAccountsJsonFile.renameTo(newFile)) {
+                if (!newFile.delete()) newFile.deleteOnExit();
+            } else {
+                if (!linkedAccountsJsonFile.delete()) linkedAccountsJsonFile.deleteOnExit();
+            }
```

Comment:
I think the file shouldn't be deleted for now in case the migration doesn't work properly for whatever reason